crates/edr_eth/src/remote/client.rs

```diff
@@ -679,8 +705,79 @@ impl RpcClient {
         .await
     }
 
+    /// Fetch the latest block number, chain id and network id in a batch call.
+    #[cfg_attr(feature = "tracing", tracing::instrument(level = "trace", skip(self)))]
+    pub async fn fetch_fork_metadata(&self) -> Result<ForkMetadata, RpcClientError> {
+        let mut inputs = vec![
+            RequestMethod::NetVersion(()),
+            RequestMethod::BlockNumber(()),
```

Comment:
Yes that's a mistake thanks! I added caching for the block number later and forgot to remove it. The tests were passing, because we're always getting the chain id from the Alchemy url, so the third return value was never utilized.

Fixed in [96602d7](https://github.com/NomicFoundation/hardhat/pull/4806/commits/96602d7138b4e8c22e67eeb05e3ea79aedffc188)